ui/pages/device_code.tsx

```diff
@@ -1,25 +1,83 @@
 import type { NextPage } from "next";
-import React from "react";
+import axios, { AxiosError } from "axios";
+import { useRouter } from "next/router";
+import React, { FormEvent, useCallback, useEffect, useState } from "react";
 import PageLayout from "../components/PageLayout";
-import { Button, Input } from "@canonical/react-components";
+import { Button, Form, Input } from "@canonical/react-components";
+
+export interface Response {
+  data: {
+    redirect_to?: string;
+  };
+}
+
+async function acceptUserCode(userCode: string, challenge: string) {
+  return axios
+    .put("/api/device?device_challenge=" + String(challenge), {
```

Comment:
```suggestion
    .put(`/api/device?device_challenge=${challenge}`, {
```
We can use the inline style here and don't need to cast to string again, I guess?